src/core/common/message.cpp

```diff
@@ -835,14 +833,12 @@ otError MessageQueue::Enqueue(Message &aMessage, QueuePosition aPosition)
     }
 
 exit:
-    return error;
+    return;
 }
 
-otError MessageQueue::Dequeue(Message &aMessage)
+void MessageQueue::Dequeue(Message &aMessage)
 {
-    otError error = OT_ERROR_NONE;
-
-    VerifyOrExit(aMessage.GetMessageQueue() == this, error = OT_ERROR_NOT_FOUND);
+    VerifyOrExit(aMessage.GetMessageQueue() == this, OT_NOOP);
```

Comment:
```suggestion
    VerifyOrExit(aMessage.IsInAQueue(), OT_NOOP);
    
    OT_ASSERT(aMessage.GetMessageQueue() == this);
```
If `aMessage` is in any queue but not the current message queue, better just let it fail in the `OT_ASSERT` below?